linker-interface/shared/src/main/scala/org/scalajs/linker/interface/LinkTimeProperties.scala

```diff
@@ -0,0 +1,64 @@
+package org.scalajs.linker.interface
+
+import org.scalajs.ir.Trees._
+
+final class LinkTimeProperties (
+  semantics: Semantics,
+  esFeatures: ESFeatures
+) {
+  import LinkTimeProperties._
+  import LinkTimeProperties.ResolvedLinkTimeValue._
+
+  private val info = "scala.scalajs.LinkingInfo"
+  private val linkTimeProperties: Map[String, ResolvedLinkTimeValue] = Map(
+    s"$info.productionMode" -> BooleanValue(semantics.productionMode),
+    s"$info.esVersion"      -> IntValue(esFeatures.esVersion.edition)
+  )
+
+  def evaluateLinkTimeCondition(cond: LinkTimeCondition): Boolean = {
+    cond match {
+      case LinkTimeCondition.Binary(op, lhs, rhs) =>
+        (resolveLinkTimeValue(lhs), resolveLinkTimeValue(rhs)) match {
```

Comment:
These should recursively call `evaluateLinkTimeCondition` in order to support arbitrary nestings of expressions, like a standard interpreter evaluation function.

It's only at the top-level that we should turn an evaluated `BooleanValue` into a `Boolean`.